tests/types/date.spec.ts

```diff
@@ -0,0 +1,74 @@
+import {StrongDate, makeDate} from '../../src/types/date';
+
+const MOCK_INITIAL = '2020';
+const MOCK_FALLBACK_DEFAULT = '1886-08';
+const MOCK_FALLBACK = '1999-10-31';
+
+describe('StrongDate', () => {
+	describe('Implementation', () => {
+		it('should set initial value to sampleInitial argument', () => {
+			const sampleInitial = '2020-12-25';
+			const result = makeDate(sampleInitial, MOCK_FALLBACK_DEFAULT);
+			expect(result()).toBe(sampleInitial);
+		});
+
+		it('should not set value when called with a string', () => {
+			const result = makeDate(null, MOCK_INITIAL);
+			const sampleValue = 'Feb 5th';
+			result(sampleValue);
+			expect(result()).toBe(MOCK_INITIAL);
+		});
+
+		it('should not set value when called with an empty string', () => {
+			const sampleFallback = '2020';
+			const emptyString = '';
+			const result = makeDate(null, sampleFallback);
+			result(emptyString);
+			expect(result()).toBe(sampleFallback);
+		});
+
+		it('should return fallback default when value is null', () => {
+			const sampleFallback = '1998';
+			const result = makeDate(null, sampleFallback);
+			expect(result()).toBe(sampleFallback);
+		});
+
+		it('should return fallback default when value is undefined', () => {
+			const sampleFallback = '1534-08';
+			const result = makeDate(undefined, sampleFallback);
+			expect(result()).toBe(sampleFallback);
+		});
+
+		it('should not set value when called with a number', () => {
+			const sampleFallback = '2020-12';
+			const numberedValue = 5 as any;
+			const result = makeDate(null, sampleFallback);
+			result(numberedValue);
+			expect(result()).toBe(sampleFallback);
+		});
+
+		it('should not set value when called with a boolean value', () => {
+			const sampleFallback = '2015';
+			const booleanValue = false as any;
+			const result = makeDate(null, sampleFallback);
+			result(booleanValue);
+			expect(result()).toBe(sampleFallback);
+		});
+	});
+
+	describe('Methods', () => {
+		describe('get', () => {
+			it('should return fallback argument when value is null', () => {
+				const sampleFallback = '1998-04-16';
+				const string = makeDate(null, MOCK_FALLBACK_DEFAULT);
+				expect(string.get(sampleFallback)).toBe(sampleFallback);
+			});
+
+			it('should return value when value is set', () => {
+				const sampleInitial = '2005-06';
+				const string = makeDate(sampleInitial, MOCK_FALLBACK_DEFAULT);
+				expect(string.get(MOCK_FALLBACK)).toBe(sampleInitial);
+			});
+		});
+	});
```

Comment:
Testing here covers get, and basic usage. The strong types package guarantees that a strong type won't set invalid types. If I set try to set a StrongString to a number, it should fail. If I try to set a negative number to a StrongUInt, it should fail.

Maintaining those guarantees over time means testing many possible cases here to validate valid URLs will set correctly and invalid URLs won't be set at all. Add a section for Test values with a ton of sample URLs and validate that the valid ones do set the correct value and the invalid ones don't change the internal value.